cli/cdsctl/template.go

```diff
@@ -105,43 +107,79 @@ var templatePushCmd = cli.Command{
 func templatePushRun(v cli.Values) error {
 	files := strings.Split(v.GetString("yaml-file"), ",")
 
-	buf := new(bytes.Buffer)
+	// create a new tar archive
 	var dir string
+	tar := new(bytes.Buffer)
 
-	// Create a new tar archive.
-	filesToRead := []string{}
-	for _, file := range files {
-		fi, err := os.Lstat(file)
+	// if the first args is an url, try to download all files
+	readFromLink := len(files) > 0 && exportentities.IsURL(files[0]) && strings.HasSuffix(files[0], ".yml")
+	if readFromLink {
+		manifestURL := files[0]
+		baseURL := manifestURL[0:strings.LastIndex(manifestURL, "/")]
+
+		// get the manifest file
+		contentFile, _, err := exportentities.OpenPath(manifestURL)
 		if err != nil {
-			fmt.Printf("Skipping file %s: %v\n", file, err)
-			continue
+			return err
+		}
+		buf := new(bytes.Buffer)
+		if _, err := buf.ReadFrom(contentFile); err != nil {
+			return fmt.Errorf("Cannot read from given remote file: %v", err)
```

Comment:
Error strings should not start with a capital letter because they'll often be prefixed before printing:

https://github.com/golang/go/wiki/Errors